src/mut_mut.rs

```diff
@@ -27,25 +23,34 @@ impl LintPass for MutMut {
 			}
 		}
 		
-		if unwrap_addr(expr).and_then(unwrap_addr).is_some() {
-			cx.span_lint(MUT_MUT, expr.span, 
-				"We're not sure what this means, so if you know, please tell us.")
-		}
+		unwrap_addr(expr).map(|e| {
+			if unwrap_addr(e).is_some() {
+				cx.span_lint(MUT_MUT, expr.span, 
+					"We're not sure what this means, so if you know, please tell us.")
+			} else {
+				match expr_ty(cx.tcx, e).sty {
+					ty_ptr(mt{ty: _, mutbl: MutMutable}) |
+					ty_rptr(_, mt{ty: _, mutbl: MutMutable}) => 
```

Comment:
The rptr apparently matches line 14 of the test. Would you suggest this should not be an error?